### PR TITLE
feat: Shareable, embeddable, beautiful recordings

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Needs https://github.com/PostHog/posthog/pull/15385

Various people have asked for embedded recordings. This could be for:
* Internal documentation 
* Using PostHog "head-lessly" to provide recordings to end users
* Sharing specific recordings with users outside the Organization

## Changes

* Modifies the SharingConfiguration to only get created when first enabled
* Adds support for Recordings as shareable items
* Improves the embeddding view
* Removes the [recently added context](https://github.com/PostHog/posthog/pull/15520) in favour of the logic approach 

## TODO
- [x] Actually make this work - need to pass the share token as an alternative to the API call
- [x] Validate the sharetoken for getting access to limited recording endpoints
- [x] Hide more UI elements when in "sharing" mode
- [x] Fix fullscreen
- [x] Hides the Person info as that should never really be shared...
- [x] Add tests
- [ ] Has a DB change so this is blocked by migration efforts

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Mostly manually although I may add some specific tests for the sharing api logic

### CODE REVIEW

**Comment by mariusandra** [comment]
I wonder if instead of making a session recording another exportable configuration, you should just make it into an insight instead?

It should be possible to easily hook this up to the data exploration system, and reuse a lot of the code.

What's needed is to create a new `SessionRecordingNode`, somewhere in [this file](https://github.com/PostHog/posthog/blob/9a24f342ab5fc6e8084958b1e03565d721d4238a/frontend/src/queries/schema.ts#L289). Something simple like 

```json
{
  "kind": "SessionRecordingNode",
  "id": "data goes here"
}
```

would suffice.

Then make sure [this file](https://github.com/PostHog/posthog/blob/0414f83888473e4b65b0d8a9c0a5a815872e1452/frontend/src/queries/Query/Query.tsx#L66) redirects to the right component, and [this file](https://github.com/PostHog/posthog/blob/c038ffb46dffff214d83d9c0e3758286db0527f2/posthog/api/query.py#L169) links to the data fetching code...

And boom, done.

For insights we have a two-tiered system, where 

```json5
{
  // all the frontend toggles go in this object
  "kind": "InsightVizNode",
  "source": {
    // all the backend filters go in this object
    "kind": "TrendsQuery",
  }
}
```

This could be used for session recordings as well, where the inside "source" node denotes the data to fetch a recording, end the outside `SessionRecordingPlayerNode` can get options like `showControls: true`, etc.

File: posthog/api/sharing.py [hunk 1]
```diff
@@ -41,55 +43,72 @@ class Meta:
...
                 raise NotFound("Dashboard not found.")
         if insight_id:
             try:
-                context["insight"] = Insight.objects.get(id=insight_id)
+                context["insight"] = Insight.objects.get(id=insight_id, team=self.team)
             except Insight.DoesNotExist:
                 raise NotFound("Insight not found.")
+        if recording_id:
+            try:
+                # TODO: These are special as we create them on the fly...
```
Comment (pauldambra) [inline posthog/api/sharing.py:77]: what's the TODO here?

**Review by pauldambra — COMMENTED** [review]
first pass over the code without running it

File: posthog/api/sharing.py (hunk 1 above)
Comment (benjackwhite) [reply posthog/api/sharing.py:77]: Should just be a NOTE. Will fix

File: posthog/api/test/test_session_recordings.py [hunk 2]
```diff
@@ -629,11 +629,41 @@ def test_regression_encoded_emojis_dont_crash(self):
...
+
+        token = self.client.patch(
+            f"/api/projects/{self.team.id}/session_recordings/{session_id}/sharing", {"enabled": True}
+        ).json()["access_token"]
+
+        self.client.logout()
+
+        # Unallowed routes
+        response = self.client.get(f"/api/projects/{self.team.id}/session_recordings/2?sharing_access_token={token}")
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply posthog/api/test/test_session_recordings.py:650]: I think thats a good idea :D

**Review by fuziontech — APPROVED** [review]
this may block two tables. But we are ok with that :shipit:

**Pull request merged**